### PR TITLE
Use bitnami chart for kafka

### PR DESCRIPTION
**Pull request by mehta-ankit**
Signed-off-by: Ankit Mehta <ankit.mehta@appian.com>

Resolves https://github.com/jaegertracing/helm-charts/issues/149

Incubator kafka chart will be obsolete soon as per these instructions: https://github.com/helm/charts#deprecation-timeline
Also liveness probe was failing for the incubator kafka chart.

Bitnami provides stable kafka chart that Jaeger's helm chart can use.

### CODE REVIEW

File: charts/jaeger/values.yaml
Comment (mehta-ankit) [inline charts/jaeger/values.yaml:94]: by default, kafka and zookeeper replica values are 1 for bitnami chart.

**Comment by mehta-ankit** [comment]
these are the values i used to test it on my cluster.
```
# Values for bitnami kafka chart: https://github.com/bitnami/charts/blob/master/bitnami/kafka/values.yaml
  kafka:
    replicaCount: 3
    podSecurityContext:
      # TIL: https://github.com/bitnami/bitnami-docker-kafka/blob/master/2/debian-10/Dockerfile#L30
      # bitnami builds its images with non-root user, but if you don't use the same user, you don't
      # have permissions to read/write on those folders on the mounted volume.
      runAsUser: 1001
      fsGroup: 1001
    resources:
      limits:
        cpu: 500m
        memory: 2Gi
      requests:
        cpu: 250m
        memory: 1Gi
    metrics:
      kafka:
        enabled: true
      jmx:
        enabled: true
    serviceAccount:
      create: true
    # Values for zookepr bitnasmi chart: https://github.com/bitnami/charts/blob/master/bitnami/zookeeper/values.yaml
    zookeeper:
      securityContext:
        enabled: true
        fsGroup: 1001
        runAsUser: 1001
      replicaCount: 3
      metrics:
        enabled: true
      serviceAccount:
        create: true
```

@naseemkullah do you think, if any of those should be default in jaeger's values.yaml file ?
The only one i could think of is `serviceAccount.create: true`. Let me know what you think.

**Comment by naseemkullah** [comment]
> these are the values i used to test it on my cluster.
> 
> ```
>   kafka:
>     replicaCount: 3
>     podSecurityContext:
>       runAsUser: 1001
>       fsGroup: 1001
>     resources:
>       limits:
>         cpu: 500m
>         memory: 2Gi
>       requests:
>         cpu: 250m
>         memory: 1Gi
>     metrics:
>       kafka:
>         enabled: true
>       jmx:
>         enabled: true
>     serviceAccount:
>       create: true
>     zookeeper:
>       securityContext:
>         enabled: true
>         fsGroup: 1001
>         runAsUser: 1001
>       replicaCount: 3
>       metrics:
>         enabled: true
>       serviceAccount:
>         create: true
> ```
> 
> @naseemkullah do you think, if any of those should be default in jaeger's values.yaml file ?
> The only one i could think of is `serviceAccount.create: true`. Let me know what you think.

Hard to say, I'd add the minimal config required for it to work out of the box and not fall over. `serviceAccount: true` does seem like a good idea for `zoopkeeper` (though it appears to be already enabled for `kafka` according to the chart).

Things like a sane securityContext would ideally be configured in the subcharts' default values. My two cents. As I do not plan to use this feature I can't really say much more.

@arpitjindal97 seeing that you originally added this feature please take a look once it is ready for review if you can!

**Comment by mehta-ankit** [comment]
Ready for review!

**Review by naseemkullah — APPROVED** [review]
lgtm, but will wait to see if @arpitjindal97 can take a look before merging

File: charts/jaeger/values.yaml
Comment (arpitjindal97) [reply charts/jaeger/values.yaml:97]: One change needed here. Although default replica set will be 1 for `kafka` and `zookeeper` but It would be better to set it explicitly. When any user wants to change this value, he/she wouldn't have to go to Bitnami's Documentation and find correct `key`

Comment (mehta-ankit) [reply charts/jaeger/values.yaml:97]: Sure. Although one could make that same argument for few other key's 😄 . I ll push that change up.

**Review by naseemkullah — APPROVED** [review]

**Comment by naseemkullah** [comment]
Thanks @mehta-ankit!

**Pull request merged**
